Center relative to parent control and fix location jitter

X9VoiD opened the pull request:
## Changes

- Center all child form or dialog relative to parent window (mostly `MainForm`)
- Attach all child form or dialog to parent window (fixes problems like losing a form/dialog behind the `MainForm` or the form/dialog not restored in conjuction with `MainForm.Show()` after getting minimized)
- Fixed `MainForm` being larger than 1366x768 or smaller screens. (Previous logic checked if ClientSize is larger than Screen before applying but ClientSize does not account for window decoration, so the window's total size is still larger than the screen)
- Apply `MainForm`'s sizing and centering logic before it launches by retrieving the screen through the cursor

InfinityGhost commented on OpenTabletDriver.UX/MainForm.cs:
```diff
@@ -18,13 +18,14 @@ namespace OpenTabletDriver.UX
         public MainForm()
             : base()
         {
+            InitializePlatform();
```
At least copy what was in the original `DesktopForm` class if you're going to outright remove it. This is called at `OnLoadComplete()` with the single run variable. Preferably, I'd like to keep the `DesktopForm` class so this is easier to maintain.

InfinityGhost reviewed (changes requested):
On GTK, dialogs are not even relatively close to being centered.
![image](https://user-images.githubusercontent.com/39218853/105557334-f8659e80-5cd9-11eb-9747-066d4ff49748.png)

InfinityGhost commented on OpenTabletDriver.UX/DesktopForm.cs:
```diff
@@ -1,66 +1,42 @@
-                OnInitializePlatform(e);
-            }
+            Location = new Point((int)(center.X - (ClientSize.Width / 2)), (int)(center.Y - (ClientSize.Height / 2)));
         }
```
Fix this to check if the platform is GTK, window positioning is handled automatically on that platform.
```cs
        protected virtual void InitializeForm()
        {
            if (!Platform.IsGtk)
            {
                var x = Owner.Location.X + (Owner.ClientSize.Width / 2);
                var y = Owner.Location.Y + (Owner.ClientSize.Height / 2);
                var center = new PointF(x, y);

                Location = new Point((int)(center.X - (ClientSize.Width / 2)), (int)(center.Y - (ClientSize.Height / 2)));
            }
        }
 ```

X9VoiD commented:
Fixed, and rebased to https://github.com/OpenTabletDriver/OpenTabletDriver/commit/2ddf0a75b455996ac3eba163a280c83c11b7ba4a.

![image](https://user-images.githubusercontent.com/10338031/140595472-50137a8d-906f-45f4-8e6e-edefb0cdbf0f.png)

It was a bug in Eto where `ClientSize` will always be zero until form is shown, all calls to `ClientSize._set` are ignored in Gtk as long as `Form.Show()` is not called yet

gonX commented:
CodeFactor is crying - also, rebase and/or close? No idea if this PR is still interesting

InfinityGhost commented:
Resolve conflicts and I'll test then merge.

InfinityGhost commented:
Tablet debugger size appears entirely wrong on GTK, horizontal scrollbar visible.

gonX commented on OpenTabletDriver.UX/Windows/Tablet/TabletDebugger.cs:
```diff
@@ -19,8 +19,10 @@ namespace OpenTabletDriver.UX.Windows.Tablet
+            : base(Application.Instance.MainForm)
         {
             Title = "Tablet Debugger";
+            this.ClientSize = new Size(640, 640);
```
Removing this line makes Tablet debugger size properly by default
```suggestion
```

X9VoiD replied:
Removing this may also cause issues with platforms that rely on pre-determined size for jitter-less window creation (Windows, MacOS), but will add that change for now for the sake of moving things forward.

gonX replied:
The tablet debugger already has a default width height and height specified if I'm not mistaken: https://github.com/OpenTabletDriver/OpenTabletDriver/blob/master/OpenTabletDriver.UX/Windows/Tablet/TabletDebugger.cs#L122-L123:
```cs
            #code snipped for brevity
            var splitter = new Splitter
            {
                Width = 660,
                Height = 800
            };
            this.Content = new Scrollable
            {
                Content = splitter
            };
```

gonX reviewed (changes requested):
Just 1 change to fix the issue reported by Ghost, otherwise LGTM

InfinityGhost reviewed (changes requested):
Form windows are not centered when they are created on GTK, they're slightly offset lower than they should be. ideally, they should be handled by the window manager and should not be positioned by us on Linux altogether.
